code/modules/power/cable.dm

```diff
@@ -549,32 +586,173 @@ GLOBAL_LIST_INIT(cable_coil_recipes, list(new/datum/stack_recipe("cable restrain
-				first = FALSE
-				PN = C.powernet
-				continue
-			propagate_network(C, PN)
+/obj/effect/node/layer1
+	color = "red"
+	icon_state = "l1-1-2-4-8-node"
+	vis_flags = VIS_INHERIT_ID|VIS_INHERIT_PLANE|VIS_INHERIT_LAYER|VIS_UNDERLAY
+
+/obj/effect/node/layer2
+	color = "yellow"
+	icon_state = "l2-1-2-4-8-node"
+	vis_flags = VIS_INHERIT_ID|VIS_INHERIT_PLANE|VIS_INHERIT_LAYER|VIS_UNDERLAY
+
+/obj/effect/node/layer3
+	color = "blue"
+	icon_state = "l4-1-2-4-8-node"
+	vis_flags = VIS_INHERIT_ID|VIS_INHERIT_PLANE|VIS_INHERIT_LAYER|VIS_UNDERLAY
+
+/obj/structure/cable/multilayer/update_icon_state()
+	return
 
-/obj/structure/cable_bridge/wirecutter_act(mob/living/user, obj/item/I)
+/obj/structure/cable/multilayer/update_icon()
+	var/R = 50
+	var/G = 50
+	var/B = 50
+
+	R += cable_layer & CABLE_LAYER_1 ? 150:0
+	G += cable_layer & CABLE_LAYER_2 ? 150:0
+	B += cable_layer & CABLE_LAYER_3 ? 150:0
+
+	color = rgb(R, G, B)
+
+	machinery_node?.alpha = machinery_layer & MACHINERY_LAYER_1 ? 255 : 0
+	
+	cable_node_1?.alpha = cable_layer & CABLE_LAYER_1 ? 255 : 0
+
+	cable_node_2?.alpha = cable_layer & CABLE_LAYER_2 ? 255 : 0
+
+	cable_node_3?.alpha = cable_layer & CABLE_LAYER_3 ? 255 : 0
+
+	return ..()
+
+/obj/structure/cable/multilayer/Initialize(mapload)
 	. = ..()
-	qdel(src)
+
+	var/turf/T = get_turf(src)
+	for(var/obj/structure/cable/C in T.contents - src)
+		if(C.cable_layer & cable_layer)
+			C.deconstruct()						// remove adversary cable
+	if(!mapload)
+		auto_propagate_cut_cable(src)
+	
+	machinery_node = new /obj/effect/node()
+	vis_contents += machinery_node
+	cable_node_1 = new /obj/effect/node/layer1()
+	vis_contents += cable_node_1
+	cable_node_2 = new /obj/effect/node/layer2()
+	vis_contents += cable_node_2
+	cable_node_3 = new /obj/effect/node/layer3()
+	vis_contents += cable_node_3
+	update_icon()
+
+/obj/structure/cable/multilayer/Destroy()					// called when a cable is deleted
+	QDEL_NULL(machinery_node)
+	QDEL_NULL(cable_node_1)
+	QDEL_NULL(cable_node_2)
+	QDEL_NULL(cable_node_3) 
+	return ..()									// then go ahead and delete the cable
+
+/obj/structure/cable/multilayer/examine(mob/user)
+	. += ..()
+	. += "<span class='notice'>L1:[cable_layer & CABLE_LAYER_1 ? "Connect" : "Disconnect"].</span>"
+	. += "<span class='notice'>L2:[cable_layer & CABLE_LAYER_2 ? "Connect" : "Disconnect"].</span>"
+	. += "<span class='notice'>L3:[cable_layer & CABLE_LAYER_3 ? "Connect" : "Disconnect"].</span>"
+	. += "<span class='notice'>M:[machinery_layer & MACHINERY_LAYER_1 ? "Connect" : "Disconnect"].</span>"
+
+/obj/structure/cable/multilayer/attack_hand(mob/living/user)
+	if(!user)
+		return
+	var/list/layer_list = list(
+		"Layer 1" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-red"),
+		"Layer 2" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-yellow"),
+		"Layer 3" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-blue"),
+		"Machinery" = image(icon = 'icons/obj/power.dmi', icon_state = "smes")
+		)

```

Comment:
Could you turn this into a global list instead of populating it every time the proc is called?
Maybe populate it the first time it's called and keep it cached afterwards.
Something like
```nix
if(!length(GLOB.layer_list))
	GLOB.layer_list = list(
		"Layer 1" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-red"),
		"Layer 2" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-yellow"),
		"Layer 3" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-blue"),
		"Machinery" = image(icon = 'icons/obj/power.dmi', icon_state = "smes")
		)
```